### PR TITLE
salt: add `absent` roles

### PR DESCRIPTION
**Pull request by NicolasT**

### CODE REVIEW

**Comment by NicolasT** [comment]
As an example, with a modified `bootstrap` node (modified labels):

```
$ kubectl get node bootstrap --show-labels
NAME        STATUS    ROLES            AGE       VERSION   LABELS
bootstrap   Ready     bootstrap,node   22h       v1.11.7   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,inode-role.kubernetes.io/etcd=,inode-role.kubernetes.io/master=,kubernetes.io/hostname=bootstrap,metalk8s.scality.com/version=2.0,node-role.kubernetes.io/bootstrap=,node-role.kubernetes.io/node=
```

(Note: some of the original `node-role.kubernetes.io/...` labels were renamed to `inode-role.kubernetes.io/...` for easy recovery, so those won't match as node roles)

```
$ salt-call pillar.items metalk8s:nodes:bootstrap
local:
    ----------
    metalk8s:nodes:bootstrap:
        ----------
        roles:
            - node
            - ca
            - bootstrap
        version:
            2.0

```

Finally:
```
$ salt-call state.show_top
local:
    ----------
    metalk8s-2.0:
        - metalk8s.roles.minion
        - metalk8s.roles.bootstrap
        - metalk8s.roles.ca
        - metalk8s.roles.etcd.absent
        - metalk8s.roles.master.absent
        - metalk8s.roles.node
```

**Review by TeddyAndrieux — APPROVED** [review]
Not tested but LGTM (will update my PR after this one merged)

**Comment by NicolasT** [comment]
@bert-e approve

**Pull request merged**
